module/zfs/zfs_ioctl.c

```diff
@@ -6297,7 +6301,9 @@ static void
 zfs_allow_log_destroy(void *arg)
 {
 	char *poolname = arg;
-	strfree(poolname);
+	
+	if (poolname != NULL)
+	    strfree(poolname);
```

Comment:
Thanks for proposing the patch.  The thing is `strfree(NULL)` should be safe except for very old versions of ZoL.  It calls `kfree()` which is handled properly even by 2.6.32 era kernels.  Are you able to reproduce this issue reliably?